app/src/main/java/ch/epfl/sdp/map/MapsActivity.java

```diff
@@ -120,58 +122,67 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
-                            ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
-                        ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, 101);
+                        commonDatabaseAPI.selectLobby(selectLobbyRes -> {
+                            if (selectLobbyRes.isSuccessful()) {
+                                PlayerForFirebase playerForFirebase = EntityConverter.playerToPlayerForFirebase(playerManager.getCurrentUser());
+                                Map<String, Object> data = new HashMap<>();
+                                data.put("count", playerManager.getNumPlayersBeforeJoin() + 1);
+                                if (playerManager.isServer()) data.put("startGame", false);
+                                Log.d("Database", "Lobby selected:" + playerManager.getLobbyDocumentName());
+                                joinLobby(playerForFirebase, data);
+                            } else {
+                                Toast.makeText(MapsActivity.this, selectLobbyRes.getException().getMessage(), Toast.LENGTH_LONG).show();
+                            }
+                        });
                     } else {
-                        locationFinder = new GoogleLocationFinder((LocationManager) getSystemService(Context.LOCATION_SERVICE));
+                        Toast.makeText(MapsActivity.this, fetchUserRes.getException().getMessage(), Toast.LENGTH_LONG).show();
                     }
-
-                    commonDatabaseAPI.selectLobby(selectLobbyRes -> {
-                        if (selectLobbyRes.isSuccessful()) {
-                            PlayerForFirebase playerForFirebase = EntityConverter.playerToPlayerForFirebase(playerManager.getCurrentUser());
-                            Map<String, Object> data = new HashMap<>();
-                            data.put("count", playerManager.getNumPlayersBeforeJoin() + 1);
-                            if (playerManager.isServer()) data.put("startGame", false);
-                            Log.d("Database", "Lobby selected:" + playerManager.getLobbyDocumentName());
-                            joinLobby(playerForFirebase, data);
-                        } else {
-                            Toast.makeText(MapsActivity.this, selectLobbyRes.getException().getMessage(), Toast.LENGTH_LONG).show();
-                        }
-                    });
-                } else {
-                    Toast.makeText(MapsActivity.this, fetchUserRes.getException().getMessage(), Toast.LENGTH_LONG).show();
-                }
-            });
-        }
+                });
+            }
 
         Log.d("Database", "Quit map ready");
     }
 
+    // for manually starting the remote game engine (server) for tests
+    public void createAndRunServer(){

```

Comment:
Don't forget ctrl+alt+L :)

Reply:
Ctrl+alt+L? Yes and don't forget to hit save either (ctrl+s)

Reply:
It's for formatting